src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala

```diff
@@ -147,15 +171,25 @@ object DataTypes {
       for {
         invalidNonExistingFields <- nonExistingFields
         invalidAggregationFieldForTypes <- invalidTypeAggregationField
+        updatedQuery <- replaceTimestampInPredicates(entity, query, tezosPlatformDiscovery)
       } yield {
         invalidNonExistingFields ::: invalidAggregationFieldForTypes match {
-          case Nil => Right(query)
+          case Nil => Right(updatedQuery)
           case wrongFields => Left(wrongFields)
         }
       }
+
     }
   }
 
+  private def dateToIso(date: Date): String = {
```

Comment:
`java.time` package provides ready-made-apis for this kind of things (I encourage everyone to study those)

```scala
import java.time.{Instant, ZoneOffset}
import java.time.format.DateTimeFormatter.ISO_INSTANT

def formatToIso(epochMillis: Long): String =
  Instant.ofEpochMilli(epochMillis)
    .atZone(ZoneOffset.UTC)
    .format(ISO_INSTANT)
```

Reply:
ok, done